firehose.c

```diff
@@ -267,14 +277,18 @@ static int firehose_configure(struct qdl_device *qdl, bool skip_storage_init, co
 	int ret;
 
 	ret = firehose_send_configure(qdl, max_payload_size, skip_storage_init, storage, &size);
-	if (ret < 0)
-		return ret;
+	if (ret < 0) {
+		fprintf(stderr, "[CONFIGURE] request failed\n");
+		return -1;
+	}
 
 	/* Retry if remote proposed different size */
 	if (size != max_payload_size) {
 		ret = firehose_send_configure(qdl, size, skip_storage_init, storage, &size);
-		if (ret < 0)
-			return ret;
+		if (ret != FIREHOSE_ACK) {
+			fprintf(stderr, "[CONFIGURE] request failed\n");
+			return -1;
```

Comment:
might as well return ret to propagate errors from 2 levels deeper

Reply:
No, because then we might return FIREHOSE_NAK from firehose_run(), I rather keep that in {0,-1} space...